lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -96,7 +96,16 @@ def store(credit_card, options={})
         add_stored_credentials(post, credit_card, options)
         add_recurring_contract(post, options)
         add_address(post, options)
-        commit('authorise', post, options)
+
+        initial_response = commit('authorise', post, options)
+
+        if initial_response.success? == false
```

Comment:
This conditional looks redundant because ```initial_response``` is returned in by the first ```if``` clause and the ```else```. Is this intentional? If it's possible for the response to be successful without a ```pspReference```, it looks like we also need to make the ```split``` null-safe.

I'm looking at this case, where ```pspReference``` is nil: https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/adyen.rb#L473

So if that null case is a possibility, I think this block could be refactored to something like:

```
if initial_response.success? && initial_response.authorization && initial_response.authorization.split('#')[2].nil? 
  failure_response(post)
else
  initial_response
end
```

Reply:
I've refactored this conditional statement to have only `if` and `else` statements to remove the redundancy.